### ISSUE TITLE
No suitable Visual Studio instances were found

### ISSUE THREAD

**Issue by efedo** [microsoft/vcpkg#26745]
**Describe the bug**
Vcpkg is unable to install any package after upgrade to VS2022, returning error "No suitable Visual Studio instances were found" I tried to manually set VCPKG_VISUAL_STUDIO_PATH, which did not help. I already have the Visual Studio English language pack installed.

Appears to be the same bug as in the following except manually setting VCPKG_VISUAL_STUDIO_PATH to the IDE path does not fix the problem :
https://github.com/microsoft/vcpkg/issues/16638
https://stackoverflow.com/questions/66978185/vcpkg-will-not-detect-visual-studio
https://stackoverflow.com/questions/66643488/vcpkg-does-not-install-cairo/66977302

**Environment**
OS: Windows 10.0.19044 Build 19044
Compiler: MSVC 14.31.31103

**To Reproduce**
1. ./vcpkg install <anything>
-or-
1. ./vcpkg env

**Failure logs**

PS D:\Programming\lib\magnum-examples> vcpkg install glew
Computing installation plan...
The following packages will be built and installed:
    glew[core]:x86-windows -> 2.1.0#10
    opengl[core]:x86-windows -> 0.0#8
Additional packages (*) will be modified to complete this operation.
No suitable Visual Studio instances were found

PS C:\Programming\lib\vcpkg-tool> vcpkg env
Could not find Visual Studio instance at C:/Program Files/Microsoft Visual Studio/2022/Community.

**Comment by efedo** [comment]
Edited with more information.

**Comment by efedo** [comment]
More details: vswhere.exe appears to be successfully locating the VS instance, so the problem appears to be in vcpkg: [vcpkg error log 1.txt](https://github.com/microsoft/vcpkg/files/9540954/vcpkg.error.log.1.txt).

Compiling the latest vcpkg-tool results in a more detailed error message. Problem appears to be specifically related to detection of x86 triplet toolset: [vcpkg error log 2.txt](https://github.com/microsoft/vcpkg/files/9540953/vcpkg.error.log.2.txt).

**Comment by efedo** [comment]
Ran out of time to properly debug. Installing a fresh vcpkg installation did not help, so I wiped and re-installed Visual Studio, vcpkg, and CMake, which resolved the problem. Before giving up, I narrowed down the failure to get_toolset(const PreBuildInfo& prebuildinfo). vswhere.exe output appears to be identical except for version numbers/dates, so it doesn't look like that's it. I suspect it may have something to do with the environment variables returned by CMake, but don't have the old and new variable values to compare.
